src/ruby/puppetserver-lib/puppet/server/config.rb

```diff
@@ -42,17 +46,79 @@ def self.initialize_puppet_server(puppet_server_config)
     end
   end
 
-  def self.ssl_context
+  def self.puppet_only_ssl_context
     # Initialize an SSLContext for use during HTTPS client requests.
     # Do this lazily due to startup-ordering issues - to give the CA
     # service time to create these files before they are referenced here.
-    unless @ssl_context
-      @ssl_context = SSLUtils.pems_to_ssl_context(
+    unless @puppet_only_ssl_context
+      @puppet_only_ssl_context = SSLUtils.pems_to_ssl_context(
           FileReader.new(Puppet[:hostcert]),
           FileReader.new(Puppet[:hostprivkey]),
           FileReader.new(Puppet[:localcacert]))
     end
-    @ssl_context
+    @puppet_only_ssl_context
+  end
+
+  def self.ssl_context
+    self.puppet_only_ssl_context
+  end
+
+  # The logging for this method strives to match the equivalent
+  # agent logging in 'puppet/ssl/ssl_provider'
```

Comment:
Is this comment meant to refer to the `load_puppet_and_system_ssl_context` method? I don't see any logging in this one.

Reply:
yeah, I pulled that out at last minute because I felt like the size of it obscured the memoization. I'll move the comment.